src/nnet3bin/nnet3-egs-augment-image.cc

```diff
@@ -106,21 +109,41 @@ void ApplyAffineTransform(MatrixBase<BaseFloat> &transform,
                 weight_21 = (r2 - r_old) * (c_old - c1),
                 weight_22 = (r_old - r1) * (c_old - c1);
       // Handle edge conditions:
-      if (r1 < 0) {
-        r1 = 0;
-        if (r2 < 0) r2 = 0;
-      }
-      if (r2 >= num_rows) {
-        r2 = num_rows - 1;
-        if (r1 >= num_rows) r1 = num_rows - 1;
-      }
-      if (c1 < 0) {
-        c1 = 0;
-        if (c2 < 0) c2 = 0;
-      }
-      if (c2 >= num_cols) {
-        c2 = num_cols - 1;
-        if (c1 >= num_cols) c1 = num_cols - 1;
+      if (fill_mode == kNearest) {
+	if (r1 < 0) {
+	  r1 = 0;
+	  if (r2 < 0) r2 = 0;
+	}
+	if (r2 >= num_rows) {
+	  r2 = num_rows - 1;
+	  if (r1 >= num_rows) r1 = num_rows - 1;
+	}
+	if (c1 < 0) {
+	  c1 = 0;
+	  if (c2 < 0) c2 = 0;
+	}
+	if (c2 >= num_cols) {
+	  c2 = num_cols - 1;
+	  if (c1 >= num_cols) c1 = num_cols - 1;
+	}
+      } else {
+	KALDI_ASSERT(fill_mode == kReflect);
+	if (r1 < 0) {
+	  r1 = - r1;
+	  if (r2 < 0) r2 = - r2;
+	}
+	if (r2 >= num_rows) {
+	  r2 = 2 * num_rows -2 - r2;
+	  if (r1 >= num_rows) r1 = 2 * num_rows - 2 - r1;
+	}
+	if (c1 < 0) {
+	  c1 = - c1;
+	  if (c2 < 0) c2 = -c2;
+	}
+	if (c2 >= num_cols) {
```

Comment:
All these `num_cols` should be changed to `height`. This issue existed in my code to. Please fix them as well.
Maybe better to rename `num_rows` to `width` to avoid future confusion.